cachito/web/api_v1.py

```diff
@@ -29,12 +33,26 @@ def get_requests():
         value exceeds configuration's CACHITO_MAX_PER_PAGE.
     :rtype: flask.Response
     """
-    # Default versbose flag to False
+    # Check if user is filtering requests by state
+    state = flask.request.args.get('state', default=None)
+    # Default verbose flag to False
     verbose = str_to_bool(flask.request.args.get('verbose', False))
     max_per_page = flask.current_app.config['CACHITO_MAX_PER_PAGE']
     # The call to `paginate` will inspect the current HTTP request for the
     # pagination parameters `page` and `per_page`.
-    pagination_query = Request.query.paginate(max_per_page=max_per_page)
+    if state:
+        state_int = RequestStateMapping.__members__[state].value
+        query = Request.query
+        state_subquery = (
+            db.session.query(RequestState.request_id,
+                             func.max(RequestState.updated).label('updated'))
+            .group_by(RequestState.request_id)
+            .filter_by(state=state_int)
+        ).subquery()
+        filtered_query = query.join(state_subquery, Request.id == state_subquery.c.request_id)
+        pagination_query = filtered_query.paginate(max_per_page=max_per_page)
```

Comment:
If you assign `query = Request.query` outside of the if block and instead of using `filtered_query`, you just overwrite the value of `query`, then you can just do the pagination once outside of the if block as `pagination_query = query.paginate(max_per_page=max_per_page)`.

This develops a pattern where the pagination is performed the same way regardless of any filters added by the user.